main/main.c

```diff
@@ -1680,11 +1680,7 @@ static ZEND_COLD void php_message_handler_for_zend(zend_long message, const void
 
 					snprintf(memory_leak_buf, 512, "Last leak repeated %lu time%s\n", leak_count, (leak_count>1?"s":""));
 				}
-#	if defined(PHP_WIN32)
-				OutputDebugString(memory_leak_buf);
```

Comment:
Perhaps it would make sense to keep this but use `IsDebuggerPresent()` or out to stderr otherwise.

Thanks.

Reply:
Thanks, didn't know about `IsDebuggerPresent()`. I think I like this idea!

Reply:
Why though? At least for leak messages it does not appear to be useful to redirect this to a debugger. Is there some historical reason why Windows has been handled differently here? Is it due to different interleaving of stdout and stderr or something?

Reply:
@nikic Seems 59a539f6ad696a53e4358fdfc36ac4965dcbc989 provides the answer; GUI mode executables don't have `stderr` etc. Guess we need to check for GUI vs. console mode.

Reply:
I was just debugging this with a small snippet. Ofc I couldn't tell why it was changed this way back then. The [doc page](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363362(v=vs.85).aspx) tells that the handling was changing a lot through the VS versions. 

Anyway, from the current stand point, the process being debugged doesn't have a console. This has led to some strange things in the past, fe [here](https://github.com/php/php-src/blob/master/win32/console.c#L95). Also, if PHP runs as an Apache module, stderr might be redirected into the Apache error log which makes it not obvious while under debugger. Basically, things might behave a different under a debugger, what i've proposed only concerns this case and debug builds. Without debugger, the output would be passed to stderr as per Christoph's intention.

Thanks.

Reply:
Thanks, Anatol! I'm going to push an improved version.